src/ConnectionManager.ts

```diff
@@ -201,18 +214,10 @@ export class ConnectionManager {
   }
 
   private handleWeb3ReactDeactivate = () => {
-    if (this.connector) {
-      this.connector.removeListener(
-        ConnectorEvent.Deactivate,
-        this.handleWeb3ReactDeactivate
-      )
-      this.connector.removeAllListeners(ConnectorEvent.Update)
-    }
-
     // Whenever the user manually disconnects the account from their wallet, the event will be
     // intercepted by this handler, calling the disconnect method.
     // Necessary to sanitize the state and prevent the continuation of a dead connection.
-    this.disconnect().catch(console.error)
+    return this.disconnect().catch(console.error)
```

Comment:
```suggestion
    try {
      await this.disconnect()
    } catch(e) {
      console.error(e)
    }
```

You can probably remove the return type from the function with this